### PR TITLE
Documentation in Markdown and Paradox

### PR DESCRIPTION
**Pull request by patriknw**
Lightbend is "standardizing" on documentation format to make it possible to have aggregated documentation and such. The choice for that is something called Paradox. It's a sbt-plugin that adds some extensions to markdown, such as code inclusion, toc, etc. You can see how it looks like in the monitoring docs: http://monitoring.lightbend.com/docs/2.0.x/home.html

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -9,3 +9,5 @@ addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.4.0")
 addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.13.0")
 
 addSbtPlugin("de.heikoseeberger" % "sbt-header" % "1.5.1")
+
+addSbtPlugin("com.typesafe.paradox" % "sbt-paradox" % "0.1.19-SNAPSHOT")
```
Comment (2m) [inline project/plugins.sbt]:
```
addSbtPlugin("com.lightbend.paradox" % "sbt-paradox" % "0.2.0")
```
is now out

**Comment by patriknw** [comment]
Updated to the released Paradox 0.2

https://github.com/lightbend/paradox

You can build the docs with `sbt docs/paradox`, and the generated site is in `docs/target/paradox/site/index.html`

**Comment by drewhk** [comment]
Looks good so far

**Comment by patriknw** [comment]
I have completed this.
@13h3r @kciesielski ok to merge?

File: build.sbt [hunk 2]
```diff
@@ -81,6 +81,22 @@ lazy val core = project
...
+    name := "akka-stream-kafka-docs",
+    paradoxTheme := Some(builtinParadoxTheme("generic")),
+    paradoxNavigationDepth := 3,
+    paradoxProperties ++= Map(
+      "akka.version" -> akkaVersion,
+      "kafka.version" -> kafkaVersion,
+      "scala.binaryVersion"              -> scalaBinaryVersion.value,
+      "scala.version"                    -> scalaVersion.value
+    )
+  )
```
Comment (ktoso) [inline build.sbt]: nice, I'm getting convinced to Paradox :)

**Comment by kciesielski** [comment]
@patriknw I need to read it carefully, I'm not sure if I'll be able today.

**Comment by patriknw** [comment]
@kciesielski Feel free to do that afterwards. I think this is a good starting point, and then we can help out to improve. It's essentially the examples we have in the readme + some descriptions.

**Comment by kciesielski** [comment]
@patriknw +1, merging then.

**Pull request merged**